trucks.js

```diff
@@ -503,6 +511,55 @@ var foodTrucks = [
 	}
 ];
 
+
+	myObject.filterByDay = function(dayName){
+		var filteredTrucks = _.filter(this.foodTrucks,function(truck){
+			return (truck.schedule.indexOf(dayName)>-1);
+		});
+		return filteredTrucks;
+	};
+
+	myObject.getTrucks = function(){
+		return this.foodTrucks;
+	};
+	
+	myObject.getTruck = function(name){
+		var i=0;
+		for(i=0;i<this.foodTrucks.length;i++){
+			if(this.foodTrucks[i].name === name){
+				return this.foodTrucks[i];
+			}
+		}
+	};
+	
+	myObject.getFoodTypes = function (){
+		//munge together all the food types from all the trucks into an array push the value 
+		var mungeArray = [];
+		var foodTypeString = '';
+		var i=0;
+		for(i=0;i<this.foodTrucks.length;i++){
+			//type is an array so have to union it in (plus that gets us unique)
+			mungeArray = _.union(mungeArray,this.foodTrucks[i].type);
+		}
+		mungeArray = _.sortBy(mungeArray);
+		// then JSON.stringify(uniqueArray) to send out the JSON list of food types
+		foodTypeString = JSON.stringify(mungeArray);
```

Comment:
This function would work fine without stringifying the foodType array you've built.